src/NUnitEngine/nunit.engine/Runners/DirectTestRunner.cs

```diff
@@ -71,7 +71,22 @@ public override TestEngineResult Explore(TestFilter filter)
             var result = new TestEngineResult();
 
             foreach (IFrameworkDriver driver in _drivers)
-                result.Add(driver.Explore(filter.Text));
+            {
+                string driverResult;
+
+                try
+                {
+                    driverResult = driver.Explore(filter.Text);
+                }
+                catch (Exception ex)
+                {
+                    if (ex is NUnitEngineException)
+                        throw;
```

Comment:
@tom-dudley Exception filters were supported in IL and VB.NET from day one, [back in 2001](https://www.ecma-international.org/publications/files/ECMA-ST-WITHDRAWN/ECMA-335,%201st%20edition,%20December%202001.pdf). I

The C# language access to this feature was added in version 6 in 2015.